profile.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package eat
+
+import (
+	"encoding/asn1"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+const (
+	// ASN1AbsoluteOIDType represents the Type number of Absolute OID ASN Encoding
+	ASN1AbsoluteOIDType = 0x06
+	// ASN1LongLenMask is used to mask bit 8 of Length Indicator byte
+	ASN1LongLenMask = 0x80
+	// ASN1LenBytesMask is used to extract first 7 bits of Length Indicator byte
+	ASN1LenBytesMask = 0x7F
+	// MaxASN1OIDLen is the constant to address the maximum OID length handled in system
+	MaxASN1OIDLen = 260
+	// MinNumOIDArcs represents the minimum required arcs for a valid OID
+	MinNumOIDArcs = 3
+)
+
+// Profile is either an absolute URI (RFC3986) or an ASN.1 Object Identifier
+type Profile struct {
+	val interface{}
+}
+
+// NewProfile instantiates a Profile object from the given input string
+// The string can either be an absolute URI or an ASN.1 Object Identifier
+// in dotted-decimal notation
+func NewProfile(urlOrOID string) (*Profile, error) {
+	p := Profile{}
+	if err := p.Set(urlOrOID); err != nil {
+		return nil, err
+	}
+	return &p, nil
+}
+
+// Set sets the internal value of the Profile object to the given urlOrOID string
+func (s *Profile) Set(urlOrOID string) error {
+	return s.decodeProfileFromString(urlOrOID)
+}
+
+// Get returns the profile as string (URI or dotted-decimal OID)
+func (s Profile) Get() (string, error) {
+	switch t := s.val.(type) {
+	case *url.URL:
+		return t.String(), nil
+	case asn1.ObjectIdentifier:
+		return t.String(), nil
+	default:
+		return "", fmt.Errorf("no valid EAT profile")
+	}
+}
+
+// IsURI checks whether a stored profile is a URI
+func (s Profile) IsURI() bool {
+	_, ok := s.val.(*url.URL)
+	return ok
+}
+
+// IsOID checks whether a stored profile is an OID
+func (s Profile) IsOID() bool {
+	_, ok := s.val.(asn1.ObjectIdentifier)
+	return ok
+}
+
+// constructASN1fromVal constructs a TLV ASN1 byte array from an ASN1 value
+// supplied as an input, assumption is a meaningful OID byte array of less than 255 bytes
+func constructASN1fromVal(val []byte) ([]byte, error) {
+	var OID [MaxASN1OIDLen]byte
+	asn1OID := OID[:2]
+	asn1OID[0] = ASN1AbsoluteOIDType
+	if len(val) < 127 {
+		asn1OID[1] = byte(len(val))
+	} else if len(val) < 256 {
```

Comment:
I think it's 256 that you want to make a `const` rather than `MaxASN1OIDLen`, because this is the real show-stopper in the implementation, right?